megatron/model/gpt_model.py

```diff
@@ -232,13 +232,13 @@ def _to_float16(inputs):
                                         tied_weight_attr='word_embeddings_weight'))
 
         if args.fp32_residual_connection:
-            if hasattr(args, 'attn_mask'):
+            if hasattr(args, 'pretrain_causal_attention') and (args.pretrain_causal_attention is True):
                 self.specs.append(lambda x: x.transpose(0, 1).contiguous().float())
             else:
                 # EmbeddingPipe returns attention mask as well
                 self.specs.append(lambda x: (x[0].transpose(0, 1).contiguous().float(), *x[1:]))
         else:
-            if hasattr(args, 'attn_mask'):
+            if hasattr(args, 'pretrain_causal_attention') and (args.pretrain_causal_attention is True):
```

Comment:
```suggestion
            if hasattr(args, 'pretrain_causal_attention') and args.pretrain_causal_attention:
```

typically one doesn't do `is True`, is there a special purpose for writing it out explicitly? same with all the edits below.

Reply:
perhaps this is dealing with 3-state and you don't want to check for `None`?

Reply:
but really, get rid of undefined case complete and set the default to be `False`. Problem solved.

Reply:
Hum okay let's go with `getattr(args, 'pretrain_causal_attention', False)` Thanks!